cdap-watchdog/src/main/java/co/cask/cdap/logging/context/MapReduceLoggingContext.java

```diff
@@ -32,11 +30,13 @@
    * @param namespaceId namespace id
    * @param applicationId application id
    * @param mapReduceId mapreduce job id
+   * @param adapterId adapter id
    */
   public MapReduceLoggingContext(final String namespaceId, final String applicationId, final String mapReduceId,
-                                 @Nullable String adapterId) {
-    super(namespaceId, applicationId, adapterId);
+                                 String adapterId) {
+    super(namespaceId, applicationId);
```

Comment:
Actually, ``adapterId`` should be nullable here and set only if non-null? That would be when a mapreduce is run outside of an adapter.